pom.xml

```diff
@@ -2040,6 +2040,27 @@
               </execution>
             </executions>
           </plugin>
+          <plugin>
+            <groupId>org.jacoco</groupId>
+            <artifactId>jacoco-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <goals>
+                  <goal>prepare-agent</goal>
+                </goals>
+              </execution>
+              <execution>
```

Comment:
Can we try adding `report-aggregate` goal somewhere to report the coverage across modules?  Right now it does not seem like the report is aggregated across modules.
```
                        <execution>
                            <id>report</id>
                            <phase>prepare-package</phase>
                            <goals>
                                <goal>report-aggregate</goal>
                            </goals>
                        </execution>
```

Reply:
This is the example: https://github.com/jacoco/jacoco/blob/master/jacoco-maven-plugin.test/it/it-report-aggregate/report/pom.xml

Reply:
I tried that and it reports nothing, probably because we use the v2 reportage plugin which is recommended, let me tweak and have a try again.

Reply:
It looks like it is very tricky to make the report-aggregate work, based on these links:
 
- https://groups.google.com/g/jacoco/c/FpdLbxsXSTY
 - https://github.com/jacoco/jacoco/blob/master/jacoco-maven-plugin.test/it/it-report-aggregate/report/pom.xml
 
 Basically if we wanna to make the aggrerator work, we should have a separate module named `hudi-reporter` that adds the correspendent denpendencies for the multiple moldules that we want to report, otherwise, we always got empty report files.
 
 But some modules are actually dinamically loaded based on the active maven compile profile, which makes the issue even more complicated.

Reply:
I took several attempts like let each sub-module have a report-aggregate gole, but that still reports 0 coverage. So revert the code back to the non-aggregate version.

Reply:
I'm wondering if the goal should be `verify`, i.e., `mvn clean verify` for `report-aggregate` to work.  Have you tried the exact repo from https://github.com/jacoco/jacoco/blob/master/jacoco-maven-plugin.test/it/it-report-aggregate and see if it works locally with maven?

For the issue of dynamic loaded modules, maybe we can first try excluding such modules like `hudi-spark` and run for `hudi-spark-client` for example.

Reply:
It turns out to be that we must run the `mvn test` in the root dir of the project and we can not specify specific modules with `-pl` option. This is a limination for our Azure CI becase we split the modules into 6 separate job.

Reply:
One idea is that we can aggregate the report for a particular module and run the `mvn test` from that module directory.  If that works, we can run tests in different modules in a different way without `-pl` as a compromise.

Reply:
Like I said, execute `mvn test` under module dir does not work.